pkg/pillar/agentlog/agentlog.go

```diff
@@ -76,6 +80,18 @@ func handleSignals(sigs chan os.Signal) {
 			case syscall.SIGUSR1:
 				stacks := getStacks(true)
 				stackArray := strings.Split(stacks, "\n\n")
+
+				sigUsr1File, err := os.OpenFile(sigUsr1FileName, os.O_WRONLY|os.O_CREATE|os.O_SYNC, 0755)
```

Comment:
I forget whether this does append by default if case you have to specific O_TRUNC to start empty.

Reply:
You do.
I don't think you need O_SYNC. So you can use overwriteFile

Reply:
Oh - you have multiple writes here.
So can't use overwriteFile

Reply:
Took care of it

Reply:
> > Doesn't mean we shouldn't make the logging pipeline reliable and usable end to end, but that seems to still be months away.
> 
> Fair enough @eriknordmark -- I am still unhappy about `spoofStdFDs` though -- the way I read it seems that we're now circumventing the normal stdout/stderr flow that feeds memlogd? How's that supposed to work @gkodali-zededa ?

@rvs Service starts with it stdout, stderr attached to pipes that terminate at containerd-shim.

stdout ----------------------------> containerd-shim-fdX
stderr ----------------------------> containerd-shim-fdY

We make a clone of stdout. New cloned FD is created such that message sent to this cloned FD also go over the same pipe that stdout is attached to.

stdout ----------------------------> containerd-shim-fdX
cloned-stdout --------------------> containerd-shim-fdX
stderr ----------------------------> containerd-shim-fdY

Now we give this cloned-stdout to logrus. Logs go to containerd-shim using cloned-stdout socket.

We then create a file FD and then copy this new file FD (properties) into stdout FD.

stdout ----------------------------> containerd-shim-fdX
cloned-stdout --------------------> containerd-shim-fdX (via logrus)
stderr ----------------------------> containerd-shim-fdY
file-fd ----------------------------> file-on-persist

becomes

stdout ----------------------------> file-on-persist
cloned-stdout --------------------> containerd-shim-fdX (via logrus)
stderr ----------------------------> containerd-shim-fdY

There are other files that we create in persist like fatal-stack, fatal-reason, sigusr1, sigusr2 to which we explicitly write from respective handlers the appropriate messages/traces when relevant events occur.

Reply:
Hey @gkodali-zededa -- did you mean:
```
stderr ----------------------------> file-on-persist
cloned-stdout --------------------> containerd-shim-fdX (via logrus)
stderr ----------------------------> containerd-shim-fdY
```

In your last diagram? I don't see where stdout gets redirected to the file in persist in the code (and @eriknordmark seems to agree that it is only stderr that we pipe to files)

Reply:
My diagram is correct. It is only stdout that we re-direct to file. Look for Dup2 system call, where we copy persist file FD props to stdout FD.
https://github.com/lf-edge/eve/blob/50c4122829be096a762cb7989b5f41dc36052f6f/pkg/pillar/agentlog/agentlog.go#L434

stderr (unexpected crash/panic traces) still reach memlogd and then end up in cloud/ES.